modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -176,7 +178,12 @@ protected void doStart() {
 
             serverBootstrap.group(new NioEventLoopGroup(workerCount, daemonThreadFactory(settings,
                 HTTP_SERVER_WORKER_THREAD_NAME_PREFIX)));
-            serverBootstrap.channel(NioServerSocketChannel.class);
+
+            if (ByteBufAllocator.DEFAULT.isDirectBufferPooled()) {
```

Comment:
IMO, this conditional (same goes for the one in `Netty4Transport`) deserves at least a one liner comment explaining that we do our own management of the direct write buffer if we don't have a pooled direct allocator.